conanfile.py

```diff
@@ -266,34 +280,41 @@ def _openssl_path(self):
             pck = "OpenSSL"
         return self.conanfile.deps_cpp_info[pck].rootpath
 
-    def _build(self, dest_dir):
-        from conan.tools.gnu import AutotoolsToolchain, Autotools
-
+    def generate(self, dest_dir):
+        from conan.tools.gnu import AutotoolsToolchain, AutotoolsDeps
         tc = AutotoolsToolchain(self.conanfile, prefix=dest_dir)
         tc.configure_args.extend(["--enable-shared", f"--with-openssl={self._openssl_path}"])
-        tc.make_args.append("-j8")
+        tc.generate()
 
+        tc = AutotoolsDeps(self.conanfile)
         # We need to do this manually because `AutotoolsDeps` doesn't add `tools_requires` deps
         for dep in self.conanfile.deps_cpp_info.deps:
             info = self.conanfile.deps_cpp_info[dep]
-            tc.extra_cflags.extend([f"-I{x}" for x in info.include_paths])
-            tc.extra_ldflags.extend([f"-L{x}" for x in info.lib_paths])
+            for x in info.include_paths:
+                tc.environment.append("CPPFLAGS", f"-I{x}")
+            for x in info.lib_paths:
+                tc.environment.append("LDFLAGS", f"-L{x}")
```

Comment:
I think that you can keep the nicer syntax since `Environment.append` accepts lists:
```py
tc.environment.append("CPPFLAGS", [f"-I{x}" for x in info.include_paths])
tc.environment.append("LDFLAGS", [f"-L{x}" for x in info.lib_paths])
```